app/api/server/v1/im.js

```diff
@@ -32,22 +32,53 @@ function findDirectMessageRoom(params, user) {
 
 API.v1.addRoute(['dm.create', 'im.create'], { authRequired: true }, {
 	post() {
-		const { username, usernames } = this.requestParams();
+		const { username, usernames, excludeSelf } = this.requestParams();
 
 		const users = username ? [username] : usernames && usernames.split(',').map((username) => username.trim());
+		const selfExcluded = excludeSelf && hasPermission(this.userId, 'view-room-administration');
 
 		if (!users) {
 			throw new Meteor.Error('error-room-not-found', 'The required "username" or "usernames" param provided does not match any direct message');
 		}
 
-		const room = Meteor.call('createDirectMessage', ...users);
+		const room = Meteor.call(excludeSelf ? 'createDirectMessageForOthers' : 'createDirectMessage', ...users);
```

Comment:
as I commented below, since you have already created a function for this, I'd recommend to change this `Meteor.call` for a regular function call for `createDirectMessage` function.